.werft/util/werft.ts

```diff
@@ -36,29 +38,49 @@ export class Werft {
 
     public phase(name, desc?: string) {
         // When you start a new phase the previous phase is implicitly closed.
-        if (this.currentPhaseSpan) {
-            this.endPhase()
+        if (this.phases[this.currentPhase] && !this.phases[this.currentPhase].closed) {
+            console.log(`WARNING: The phase "${name}" is starting although the previous phase "${this.currentPhase}" is still active.`);
+        }
+
+        // Check if the phase already exists
+        if (this.phases[name]) {
+            throw new Error(`The phase "${name}" already exists!`)
         }
 
         const rootSpanCtx = trace.setSpan(context.active(), this.rootSpan);
-        this.currentPhaseSpan = this.tracer.startSpan(`phase: ${name}`, {
+        const phaseSpan = this.tracer.startSpan(`phase: ${name}`, {
             attributes: {
                 'werft.phase.name': name,
                 'werft.phase.description': desc
             }
         }, rootSpanCtx)
-        this.currentPhaseSpan.setAttributes(this.globalSpanAttributes)
+
+        this.phases[name] = { "span": phaseSpan, "closed": false, "slices": { } };
+
+        this.phases[name].span.setAttributes(this.globalSpanAttributes)
```

Comment:
I think this can be simplified by calling .setAttributes on `phaseSpan` instead of calling it on `this.phases[name]`.